ynh_add_config/ynh_add_config

```diff
@@ -0,0 +1,135 @@
+
+    cp "$template_path" "$destination"
+
+    ynh_replace_vars --file="$destination"
+
+    ynh_store_file_checksum --file="$destination"
+}
+
+# Replace variables in a file
+#
+# usage: ynh_replace_vars --file="file"
+# | arg: -f, --file=     - File where to replace variables
+#
+# The helper will replace the following keywords with global variables
+# that should be defined before calling this helper :
+#   __PATH__      by  $path_url
+#   __NAME__    by $app
+#   __NAMETOCHANGE__    by $app
+#   __USER__    by $app
+#   __FINALPATH__    by $final_path
+#   __PHPVERSION__    by $YNH_PHP_VERSION
+#
+# And any dynamic variables that should be defined before calling this helper like:
+#   __DOMAIN__    by $domain
+#   __APP__    by $app
+#   __VAR_1__    by $var_1
+#   __VAR_2__    by $var_2
+#
+#
+ynh_replace_vars () {
+    # Declare an array to define the options of this helper.
+    local legacy_args=f
+    local -A args_array=( [f]=file= )
+    local file
+    # Manage arguments with getopts
+    ynh_handle_getopts_args "$@"
+
+    # Replace specific YunoHost variables
+    if test -n "${path_url:-}"
+    then
+        # path_url_slash_less is path_url, or a blank value if path_url is only '/'
+        local path_url_slash_less=${path_url%/}
+        ynh_replace_string --match_string="__PATH__/" --replace_string="$path_url_slash_less/" --target_file="$file"
+        ynh_replace_string --match_string="__PATH__" --replace_string="$path_url" --target_file="$file"
+    fi
+    if test -n "${app:-}"; then
+        ynh_replace_string --match_string="__NAME__" --replace_string="$app" --target_file="$file"
+        ynh_replace_string --match_string="__NAMETOCHANGE__" --replace_string="$app" --target_file="$file"
+        ynh_replace_string --match_string="__USER__" --replace_string="$app" --target_file="$file"
+    fi
+    if test -n "${final_path:-}"; then
+        ynh_replace_string --match_string="__FINALPATH__" --replace_string="$final_path" --target_file="$file"
+    fi
+    if test -n "${YNH_PHP_VERSION:-}"; then
+        ynh_replace_string --match_string="__PHPVERSION__" --replace_string="$YNH_PHP_VERSION" --target_file="$file"
+    fi
+
+    # Replace othes variables
+
+    # List other unique (__ __) variables in $file
+    local uniques_vars=( $(grep -o '__[^.]*__' $file | sort --unique | sed "s@__\([^.]*\)__@\L\1@g" ))
+
+    # Do the replacement
+    local delimit=@
+    for one_var in "${uniques_vars[@]}"
+    do
+        if test -n "${!one_var:-}"; then
+            match_string="__${one_var^^}__"
+            match_string=${match_string//${delimit}/"\\${delimit}"}
+            replace_string="${!one_var}"
+            replace_string=${replace_string//${delimit}/"\\${delimit}"}
+            sed --in-place "s${delimit}${match_string}${delimit}${replace_string}${delimit}g" "$file"
+        else
+            ynh_die --message="\$$one_var wasn't initialized when trying to replace __${one_var^^}__ in $file"
+        fi
```

Comment:
```suggestion
        # Validate that one_var is indeed defined
        test -n "${!one_var:-}" || ynh_die --message="\$$one_var wasn't initialized when trying to replace __${one_var^^}__ in $file"
        
        # Escape delimiter in match/replace string
        match_string="__${one_var^^}__"
        match_string=${match_string//${delimit}/"\\${delimit}"}
        replace_string="${!one_var}"
        replace_string=${replace_string//${delimit}/"\\${delimit}"}
        
        # Actually replace (sed is used instead of ynh_replace_string to avoid triggering an epic amount of debug logs)
        sed --in-place "s${delimit}${match_string}${delimit}${replace_string}${delimit}g" "$file"
```